lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -565,9 +565,12 @@ class TTIRLayoutDPSOperandsRewriter
 
     assert(op->template hasTrait<TTIROp::Trait>());
     bool modified = false;
+    int idx = -1;
     for (auto &operand : op->getOpOperands()) {
+      idx++;
       bool isResult = op.isDpsInit(&operand);
-
+      if (idx == 0 or idx == 1 or idx == 2)
```

Comment:
nit: `if (idx >=0 and idx <= 2)`

Reply:
And please use C++ logical operators such as `&&, ||`.

Reply:
Ah sorry, my bad, guess I'm too used to python now :)

`if (idx >=0 && idx <= 2)`

Reply:
Fixed this